### ISSUE TITLE
[BUG] `torchrl.objectives.SACLoss` internal objective does not mask out terminated states

### ISSUE THREAD

**Issue by fmeirinhos** [pytorch/rl#2590]
## Describe the bug

In the `torchrl.objectives.SACLoss` the `_compute_target_v2` method fails when the next observations include `terminated` or `done` states. Specifically, [the function attempts to compute actions for these done states](https://github.com/pytorch/rl/blob/c0ba3ff54c433314e140d0b5f72e0011bfdfed50/torchrl/objectives/sac.py#L715) without masking them out, possibly leading to errors.

Implementing a mask to exclude terminated states from the action computation should prevent this issue.

## To Reproduce

This is the same example given in the docstring, but with an agent that breaks down on terminated states (which are assumed to have nan observations)

```python
import torch
from torch import nn
from torchrl.data import Bounded
from tensordict import TensorDict
from torchrl.modules.distributions import NormalParamExtractor, TanhNormal
from torchrl.modules.tensordict_module.actors import ProbabilisticActor, ValueOperator
from torchrl.modules.tensordict_module.common import SafeModule
from torchrl.objectives.sac import SACLoss

_ = torch.manual_seed(42)
n_act, n_obs = 4, 3
spec = Bounded(-torch.ones(n_act), torch.ones(n_act), (n_act,))

class ErrorOnNaNObservationModule(nn.Module):
    """
    Custom Actor Module that raises an error when 'next.observation' contains NaNs
    This is a way to simulate an agent that can't act on terminated / done states
    """
    def __init__(self):
        super().__init__()
        self.original_module = nn.Sequential(nn.Linear(n_obs, 2 * n_act), NormalParamExtractor())

    def forward(self, observation):
        if torch.isnan(observation).any():
            raise ValueError("observation containing NaNs and cannot compute an action.")
        
        return self.original_module(tensordict)


safe_module = SafeModule(
    module=ErrorOnNaNObservationModule(),
    in_keys=["observation"],
    out_keys=["loc", "scale"],
)
actor = ProbabilisticActor(
    module=safe_module,
    in_keys=["loc", "scale"],
    spec=spec,
    distribution_class=TanhNormal,
)


class ValueClass(nn.Module):
    def __init__(self):
        super().__init__()
        self.linear = nn.Linear(n_obs + n_act, 1)

    def forward(self, obs, act):
        return self.linear(torch.cat([obs, act], -1))


qvalue = ValueOperator(
    module=ValueClass(),
    in_keys=["observation", "action"],
)
value = ValueOperator(
    module=nn.Linear(n_obs, 1),
    in_keys=["observation"],
)
loss = SACLoss(actor, qvalue)
batch = [2]
action = spec.rand(batch)
data = TensorDict(
    {
        "observation": torch.randn(*batch, n_obs),
        "action": action,
        ("next", "done"): torch.tensor([False, True]).unsqueeze(-1),          # Second sample is done
        ("next", "terminated"): torch.tensor([False, True]).unsqueeze(-1),    # Second sample is terminated
        ("next", "reward"): torch.randn(*batch, 1),
        ("next", "observation"): torch.randn(*batch, n_obs),
    },
    batch,
)

# Introduce NaN in the 'next.observation' for the terminated state
# This will trigger the agent into trying to perform an action in a terminated / done state
data["next"]["observation"][1] = float('nan')

loss(data)
```

## Checklist

- [x] I have checked that there is no similar issue in the repo (**required**)
- [x] I have read the [documentation](https://github.com/pytorch/rl/tree/main/docs/) (**required**)
- [x] I have provided a minimal working example to reproduce the bug (**required**)

**Comment by vmoens** [comment]
Closed by #2606
